dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -176,10 +182,49 @@ private void cacheInit() {
                 if (cacheStoreService.supportsStoreByReference)
                     config = config.setStoreByValue(false);
                 try {
-                    if (trace && tc.isDebugEnabled())
-                        tcInvoke(cacheStoreService.tcCacheManager, "createCache", metaCacheName, config);
+                    Class<?> cachingProviderClass = cacheStoreService.cacheManager.getClass();
+                    if ("org.infinispan.jcache.remote.JCacheManager".equals(cachingProviderClass.getName())) {
+                        if (trace && tc.isDebugEnabled())
+                            tcInvoke(cacheStoreService.tcCacheManager, "createCache", metaCacheName, config);
+                        
+                        // Get RemoteCacheManager cm field from remote JCacheManager
+                        Field cmField = cachingProviderClass
+                                        .getDeclaredField("cm");
+                        cmField.setAccessible(true);
```

Comment:
Requires doPrivileged. Is it already on the call stack?

Reply:
Yes the call stack includes:

`        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
            cacheInit();
            return null;
        });`